cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -384,13 +384,13 @@ public long getApplicationCount() throws IOException {
 
   @Nullable
   public ApplicationMeta getLatest(ApplicationReference appReference) throws IOException {
-    Range range = getNamespaceAndApplicationRange(appReference.getNamespace(), appReference.getApplication());
     // scan based on: latest field set to true
-    Field<?> indexField = Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true");
+    Range range = getLatestApplicationRange(appReference);
     StructuredTable appSpecTable = getApplicationSpecificationTable();
 
+    // Get the most recently created latest version
     try (CloseableIterator<StructuredRow> iterator =
-           appSpecTable.scan(range, 1, Collections.singletonList(indexField))) {
+           appSpecTable.scan(range, 1, StoreDefinition.AppMetadataStore.CREATION_TIME_FIELD, SortOrder.DESC)) {
```

Comment:
This scan essentially has the same effect as before, but provides a more accurate guarantee